src/test/kotlin/com/ampnet/reportservice/controller/AdminControllerTest.kt

```diff
@@ -65,7 +66,34 @@ class AdminControllerTest : ControllerTestBase() {
 
             val pdfContent = result.response.contentAsByteArray
             verifyPdfFormat(pdfContent)
-            // File(getDownloadDirectory(usersAccountsSummary)).writeBytes(pdfContent)
+            // File(getDownloadDirectory("usersAccountsSummary.pdf")).writeBytes(pdfContent)
+        }
+    }
+
+    @Test
+    @WithMockCrowdfundUser(privileges = [PrivilegeType.PRA_PROFILE])
+    fun mustBeAbleToDownloadXlsxForVerifiedUsers() {
+        suppose("User service will return a list of users") {
+            val user = createUserExtendedResponse(userUuid, role = Role.PLATFORM_MANAGER)
+            val secondUser = createUserExtendedResponse(secondUserUuid)
+            val thirdUser = createUserExtendedResponse(thirdUserUuid)
+            val coopResponse = createCoopResponse()
+            val response = createUsersExtendedResponse(
+                listOf(user, secondUser, thirdUser), coopResponse
+            )
+            BDDMockito.given(userService.getAllActiveUsers(coop)).willReturn(response)
+        }
+
+        verify("Platform manager can get xlsx report") {
+            val result = mockMvc.perform(
+                MockMvcRequestBuilders.get("/admin/report/xlsx")
+                    .param("type", XlsxType.VERIFIED.name)
```

Comment:
It would be more suitable for the docs for the request param to be in lowercase.

Reply:
@mightymatth what is your comment/requirement on `type` value lower/upper case?

Reply:
```
// yes for me
GET /api/reports/users/verified
GET /api/reports/users?type=verified
GET /api/reports/users?type=VERIFIED

// no for me
GET /api/reports/users/VERIFIED
```

There's a difference between whether the value is a **query parameter** or a **segment** (part of the actual route).
If it's a query parameter, I don't care about the case, because the value can be anything (it can stay in uppercase if you ask me). When we talk about segments, it should be in lowercase because we break the convention. I'd really not be happy to see `GET /Api/reports/USERS/VerIFIED` in my code...

So, when deciding whether something is a query parameter or segment, ask yourself:
1. is the value fixed? (hint: segment, lowercase),
2. is the value arbitrary? (hint: query param, anycase. I mean, anything that can be URL encoded).

That's how I see it, but it's not a Bible.